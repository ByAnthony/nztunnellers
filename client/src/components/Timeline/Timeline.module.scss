@use 'src/index.scss';

.timeline {
  margin-bottom: index.$m-margin;
  font-weight: 600;
  color: index.$text-color;
  text-align: left;

  p {
    margin: 0;
    font-size: 1rem;
  }

  @include index.component-margin;

  .back {
    margin: 0 0 index.$m-margin 0;
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: center;
    color: index.$text-color;

    .profile {
      font-size: 1.1rem;
      font-weight: 500;
    }
  }

  .line {
    border-left: 1px solid index.$primary-light-color;
    margin-left: index.$m-margin;
  }

  .date {
    margin-left: -10px;
    padding: index.$xs-margin 0;
    background-color: index.$primary-dark-color;

    .year {
      font-size: 1.15rem;
      font-weight: 600;
    }

    .day-month {
      font-size: 1.1rem;
      font-weight: 300;
    }
  }

  .event-wrapper {
    margin: index.$s-margin 0;
  }

  .main-event {
    @include index.fullwidth-main-card;

    margin-left: -25px;

    .disease {
      display: block;
      padding-bottom: index.$xs-margin;
      font-size: 1rem;
      font-weight: 300;
    }
  }

  .company-event {
    margin-left: -25px;
    border-radius: index.$border-radius;
    border: 1px solid index.$primary-light-color;
    background-color: index.$primary-medium-color;

    p {
      padding: index.$s-margin;
    }

    img {
      max-width: calc(100% - 1px);
      width: 100%;
      border: 1px solid index.$primary-light-color;
      border-top-right-radius: index.$border-radius;
      border-top-left-radius: index.$border-radius;
      filter: grayscale(100%);
    }
  }

  .tunneller-event-with-title {
    @include index.fullwidth-main-card;

    margin-left: -10px;
    border-bottom: 1px solid index.$primary-light-color;
    background-color: index.$primary-dark-color;
  }

  .tunneller-event {
    @include index.fullwidth-secondary-card;

    margin-left: -10px;
    border-bottom: 1px solid index.$primary-light-color;
    background-color: index.$primary-dark-color;

    p {
      font-size: 1rem;
    }

    span {
      font-weight: 300;
    }
  }
}
