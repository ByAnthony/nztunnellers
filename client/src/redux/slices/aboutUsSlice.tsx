import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { AboutUs } from '../../types/about-us';

export const aboutUsApi = createApi({
  reducerPath: 'aboutUsApi',
  baseQuery: fetchBaseQuery({
    baseUrl: 'http://localhost:5000/',
  }),
  endpoints: (builder) => ({
    getAboutUs: builder.query<AboutUs, void>({
      query: () => ({
        url: '/about-us',
      }),
    }),
  }),
});

export const {
  useGetAboutUsQuery,
} = aboutUsApi;
