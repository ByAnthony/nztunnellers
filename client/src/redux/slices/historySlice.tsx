import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { Article, Articles } from '../../types/article';

export const historyApi = createApi({
  reducerPath: 'historyApi',
  baseQuery: fetchBaseQuery({
    baseUrl: 'http://localhost:5000/',
  }),
  endpoints: (builder) => ({
    getAllHistoryArticleLink: builder.query<Array<Articles>, void>({
      query: () => ({
        url: '/',
      }),
    }),
    getHistoryArticleById: builder.query<Article, string>({
      query: (id: string) => `history/${id}`,
    }),
  }),
});

export const {
  useGetAllHistoryArticleLinkQuery,
  useGetHistoryArticleByIdQuery,
} = historyApi;
