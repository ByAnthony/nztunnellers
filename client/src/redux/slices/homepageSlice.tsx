import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { Articles } from '../../types/article';

export const homepageApi = createApi({
  reducerPath: 'homepageApi',
  baseQuery: fetchBaseQuery({
    baseUrl: 'http://localhost:5000/',
  }),
  endpoints: (builder) => ({
    getHomepageData: builder.query<Record<string, Array<Articles>>, void>({
      query: () => ({
        url: '/',
      }),
    }),
  }),
});

export const {
  useGetHomepageDataQuery,
} = homepageApi;
