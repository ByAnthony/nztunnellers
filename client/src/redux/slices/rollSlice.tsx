import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { Details } from '../../types/roll';
import { Profile } from '../../types/tunneller';

export const tunnellersApi = createApi({
  reducerPath: 'tunnellersApi',
  baseQuery: fetchBaseQuery({
    baseUrl: 'http://localhost:5000/',
  }),
  endpoints: (builder) => ({
    getAllTunnellers: builder.query<Record<string, Array<Details>>, void>({
      query: () => ({
        url: 'roll/',
      }),
    }),
    getTunnellerById: builder.query<Profile, number>({
      query: (id: number) => `roll/${id}`,
    }),
  }),
});

export const { useGetAllTunnellersQuery, useGetTunnellerByIdQuery } = tunnellersApi;
