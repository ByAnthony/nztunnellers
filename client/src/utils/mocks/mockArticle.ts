import { AboutUs } from '../../types/about-us';
import {
  Article, Image, Next, Section,
} from '../../types/article';

export const mockTitle: string = 'My Awesome\\Article Title';

export const mockChapter: number = 1;

export const mockTopImage: Image = {
  file: 'img-123.png',
  alt: 'Accessible alt text',
};

export const mockArticleImage: Image = {
  ...mockTopImage,
  title: 'An Amazing Photo',
  photographer: 'The French Photographer Doisneau',
  reference: 'Where the photo is preserved',
};

export const mockSectionOne: Section = {
  title: 'Section Title',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea *commodo consequat*. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\nLorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat[1](#footnote_1). Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.\\n\\n(6--July 2023)',
};

export const mockSectionTwo: Section = {
  title: 'Section Title',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea *commodo consequat*. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\nLorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat[1](#footnote_1). Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.\\n\\n(6--July 2023)',
};

export const mockNextButton: Next = {
  url: 'my-path-to-next-chapter/',
  title: 'Next\\Chapter',
  chapter: 3,
};

export const mockNotes: string = '[1.](#reference_1) Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\n[2.](#reference_2) Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.';

export const mockArticle: Article = {
  id: 'my-awesome-article',
  chapter: mockChapter,
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage, mockArticleImage],
  next: mockNextButton,
  notes: mockNotes,
};

export const mockAboutUs: AboutUs = {
  id: 'my-awesome-article',
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage],
};
