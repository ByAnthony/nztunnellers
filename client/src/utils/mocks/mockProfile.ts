// import { Profile } from '../../types/tunneller';
// import { mockSummary } from './mockSummary';

export const mockId = 26;

// export const mockProfile = (overrides: Partial<Profile> | undefined = undefined): Profile => ({
//   id: mockId,
//   summary: mockSummary(),
//   origins: mockOrigins(),
//   preWarYears: mockPreWarYears(),
//   militaryYears: mockMilitaryYears(),
//   postServiceYears: mockPostServiceYears(),
//   sources: mockSources(),
//   image: mockImage(),
//   ...overrides,
// });
