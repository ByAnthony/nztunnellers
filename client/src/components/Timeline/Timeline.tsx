import { useParams } from 'react-router-dom';
import { Title } from '../Title/Title';
import { HowToCite } from '../HowToCite/HowToCite';
import { today } from '../../utils/date';
import { useGetTunnellerByIdQuery } from '../../redux/slices/rollSlice';

import { Footer } from '../Footer/Footer';
import { Menu } from '../Menu/Menu';
import { TimelineEvents } from './TimelineEvents/TimelineEvents';

import STYLES from './Timeline.module.scss';

export function Timeline() {
  const { id } = useParams();
  const tunnellerId = Number(id);
  const {
    data, error, isLoading, isSuccess,
  } = useGetTunnellerByIdQuery(tunnellerId);

  if (data) {
    return (
      <>
        {error && (
        <div className={STYLES.timeline}>
          <p>An error occured</p>
        </div>
        )}
        { isLoading }
        <Menu />
        { isSuccess && (
          <>
            <div className={STYLES.timeline}>
              <div className={STYLES.header}>
                <div className={STYLES.link}>
                  <a href="/tunnellers">Tunnellers</a>
                  <span>/</span>
                  <a href={`/tunnellers/${tunnellerId}`}>{`${data.summary.name.forename} ${data.summary.name.surname}`}</a>
                </div>
                <div className={STYLES['main-title']}>
                  <Title title={'World War I\\Timeline'} />
                </div>
              </div>
              <div className={STYLES.events}>
                <div className={STYLES.line}>
                  <TimelineEvents
                    militaryYears={data.militaryYears}
                    postServiceYears={data.postServiceYears}
                  />
                </div>
              </div>
            </div>
            <HowToCite id={tunnellerId} summary={data.summary} today={today} timeline />
          </>
        )}
        <Footer />
      </>
    );
  }
  return null;
}
