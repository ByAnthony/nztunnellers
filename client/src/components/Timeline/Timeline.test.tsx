import { render } from '@testing-library/react';

import { mockProfile } from '../../utils/mocks/mockProfile';
import { useGetTunnellerByIdQuery } from '../../redux/slices/rollSlice';
import { Timeline } from './Timeline';

jest.mock('../../redux/slices/rollSlice', () => ({
  useGetTunnellerByIdQuery: jest.fn(),
}));

jest.mock('../../utils/date', () => ({
  today: new Date('2023-05-04'),
}));

test('renders profile when data is available', () => {
  (useGetTunnellerByIdQuery as jest.Mock).mockReturnValue({
    data: mockProfile,
    error: null,
    isLoading: false,
    isSuccess: true,
  });

  const { asFragment } = render(<Timeline />);

  expect(asFragment()).toMatchSnapshot();
});
