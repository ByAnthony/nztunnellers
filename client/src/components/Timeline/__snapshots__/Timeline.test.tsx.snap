// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders profile when data is available 1`] = `
<DocumentFragment>
  <div
    class="menu "
    data-testid="menu"
  >
    <a
      aria-label="Go to the Homepage"
      class="logo"
      href="/"
    >
      <img
        alt=""
        src="/nzt_logo.png"
      />
    </a>
    <div
      class="search-form"
    >
      <form>
        <input
          id="search"
          name="search"
          placeholder="Search for a Tunneller"
          type="text"
        />
      </form>
      <button
        class="search-form-button"
        type="submit"
      >
        <img
          alt=""
          height="18"
          src="/searching_hover.png"
        />
      </button>
    </div>
  </div>
  <div
    class="timeline"
  >
    <div
      class="header"
    >
      <div
        class="link"
      >
        <a
          href="/tunnellers"
        >
          Tunnellers
        </a>
        <span>
          /
        </span>
        <a
          href="/tunnellers/NaN"
        >
          John Doe
        </a>
      </div>
      <div
        class="main-title"
      >
        <div
          class="main-title"
        >
          <h1>
            <span
              class="title-line-1"
            >
              World War I
            </span>
            <span
              class="title-line-2"
            >
              Timeline
            </span>
          </h1>
        </div>
      </div>
    </div>
    <div
      class="events"
    >
      <div
        class="line"
      >
        <div>
          <h2
            aria-label="Year 1915"
            class="year-title"
          >
            1915
          </h2>
          <div>
            <div
              class="date"
            >
              <div
                class="year"
              >
                1915
              </div>
              <div
                class="day-month"
              >
                26 January
              </div>
            </div>
            <div
              class="event-wrapper"
            >
              <div
                class="main-event"
              >
                <p>
                  Enlisted
                </p>
                <span>
                  Main Body
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="date"
            >
              <div
                class="year"
              >
                1915
              </div>
              <div
                class="day-month"
              >
                14 February
              </div>
            </div>
            <div
              class="event-wrapper"
            >
              <div
                class="tunneller-event-with-title"
              >
                <p>
                  Trained
                </p>
                <span>
                  Military Camp, Somewhere
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="date"
            >
              <div
                class="year"
              >
                1915
              </div>
              <div
                class="day-month"
              >
                4 May
              </div>
            </div>
            <div
              class="event-wrapper"
            >
              <div
                class="main-event"
              >
                <p>
                  Major Event
                </p>
                <span>
                  Something happened
                </span>
              </div>
            </div>
          </div>
        </div>
        <div>
          <h2
            aria-label="Year 1916"
            class="year-title"
          >
            1916
          </h2>
          <div>
            <div
              class="date"
            >
              <div
                class="year"
              >
                1916
              </div>
              <div
                class="day-month"
              >
                19 November
              </div>
            </div>
            <div
              class="event-wrapper"
            >
              <div
                class="tunneller-event"
              >
                <p>
                  Something happened that day
                </p>
              </div>
            </div>
          </div>
          <div>
            <div
              class="date"
            >
              <div
                class="year"
              >
                1916
              </div>
              <div
                class="day-month"
              >
                12 December
              </div>
            </div>
            <div
              class="event-wrapper"
            >
              <div>
                <div
                  class="company-event"
                >
                  <img
                    alt=""
                    src="/images/roll/image.jpg"
                  />
                  <p>
                    Something happened that day for the Company
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <h2
            aria-label="Year 1917"
            class="year-title"
          >
            1917
          </h2>
          <div>
            <div
              class="date"
            >
              <div
                class="year"
              >
                1917
              </div>
              <div
                class="day-month"
              >
                9 April
              </div>
            </div>
            <div
              class="event-wrapper"
            >
              <div
                class="main-event"
              >
                <p>
                  Major Event
                </p>
                <span>
                  Something happened
                </span>
              </div>
              <div>
                <div
                  class="company-event"
                >
                  <img
                    alt=""
                    src="/images/roll/image.jpg"
                  />
                  <p>
                    Something happened that day for the Company
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="date"
            >
              <div
                class="year"
              >
                1917
              </div>
              <div
                class="day-month"
              >
                8 August
              </div>
            </div>
            <div
              class="event-wrapper"
            >
              <div
                class="tunneller-event"
              >
                <p>
                  Something happened that day
                </p>
              </div>
            </div>
          </div>
        </div>
        <div>
          <h2
            aria-label="Year 1918"
            class="year-title"
          >
            1918
          </h2>
          <div>
            <div
              class="date"
            >
              <div
                class="year"
              >
                1918
              </div>
              <div
                class="day-month"
              >
                3 March
              </div>
            </div>
            <div
              class="event-wrapper"
            >
              <div
                class="main-event"
              >
                <p>
                  Major Event
                </p>
                <span>
                  Something happened
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="date"
            >
              <div
                class="year"
              >
                1918
              </div>
              <div
                class="day-month"
              >
                13 October
              </div>
            </div>
            <div
              class="event-wrapper"
            >
              <div
                class="tunneller-event"
              >
                <p>
                  Something happened that day
                </p>
              </div>
            </div>
          </div>
        </div>
        <div>
          <h2
            aria-label="Year 1919"
            class="year-title"
          >
            1919
          </h2>
          <div>
            <div
              class="date"
            >
              <div
                class="year"
              >
                1919
              </div>
              <div
                class="day-month"
              >
                4 April
              </div>
            </div>
            <div
              class="event-wrapper"
            >
              <div
                class="main-event"
              >
                <p>
                  Major Event
                </p>
                <span>
                  Something happened
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="date"
            >
              <div
                class="year"
              >
                1919
              </div>
              <div
                class="day-month"
              >
                12 July
              </div>
            </div>
            <div
              class="event-wrapper"
            >
              <div
                class="tunneller-event"
              >
                <p>
                  Something happened that day
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="howtocite"
  >
    <h3>
      How to cite this page
    </h3>
    <p>
      Anthony Byledbal, “World War I Timeline of John Doe “,
      <em>
         New Zealand Tunnellers Website
      </em>
      , 2023 (2009), Accessed: 4 May 2023. 
      <span>
        URL: www.nztunnellers.com/history/undefined.
      </span>
    </p>
  </div>
  <div
    class="footer"
  >
    <div
      class="map"
    >
      <div
        class="links"
      >
        <div
          class="map-link"
        >
          <a
            href="/#history"
          >
            History
          </a>
        </div>
        <div
          class="map-link"
        >
          <a
            href="/tunnellers"
          >
            Tunnellers
          </a>
        </div>
        <div
          class="map-link"
        >
          <a
            href="/about-us"
          >
            About Us
          </a>
        </div>
      </div>
      <button
        aria-label="Go back to the top of the page"
        class="scroll-top"
        type="button"
      >
        ↑
      </button>
    </div>
    <div
      class="support"
    >
      <div>
        <a
          aria-label="Go to The Artois University website"
          href="https://www.univ-artois.fr/artois-university"
        >
          <img
            alt="Artois University homepage"
            src="/images/support/logo-univ-artois-blanc_0.png"
          />
        </a>
      </div>
      <div>
        <a
          aria-label="Go to The Institute for Strategic Research website"
          href="https://www.irsem.fr/en/"
        >
          <img
            alt="Institute for Strategic Research homepage"
            src="/images/support/irsem-white.png"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
